src/System.Management.Automation/engine/MshCommandRuntime.cs

```diff
@@ -2856,36 +2856,37 @@ internal void _WriteErrorSkipAllowCheck(ErrorRecord errorRecord, ActionPreferenc
                 preference = ActionPreference.Continue;
             }
 
-            switch (preference)
+            if (!(ExperimentalFeature.IsEnabled("PSNotApplyErrorActionToStderr") && isNativeError))
```

Comment:
The `errorRecord` is added to `$Error` above when `ActionPreference.SilentlyContinue == preference`. Should that be guarded in this if block as well? Maybe all the code that deals with `preference` above should be included in this if block?

Reply:
Good catch.  Will wrap all the code dealing with the pref var.